Added Emscripten __assert_fail declaration to Macros.h

Conarnar opened the pull request:
Summary: __assert_fail is declared slightly differently in the Emscripten stdlib. This may cause errors when compiling with Emscripten.

Test Plan:
N/A

Rollback Plan:

Differential Revision: D78500790

JacobSzwejbka commented on torch/headeronly/macros/Macros.h:
```diff
@@ -396,6 +396,9 @@ extern SYCL_EXTERNAL void __assert_fail(
     const char* func);
+#elif (defined(__EMSCRIPTEN__))
+// As defined in assert.h in the Emscripten stdlib
+_Noreturn void __assert_fail (const char *, const char *, int, const char *);
```
can you add the names of these vars

JacobSzwejbka replied:
I assume its similar to above and below versions of this

JacobSzwejbka commented:
Context: 

ET mirrors these types from pytorch/pytorch with the hope to one day just share the source. Conan here is working on adding emscripten support to ET so we can run in browsers.

Conarnar committed:
Added Emscripten __assert_fail declaration to Macros.h (#158580)

Summary:

__assert_fail is declared slightly differently in the Emscripten stdlib. This may cause errors when compiling with Emscripten.

Test Plan:
N/A

Rollback Plan:

Reviewed By: JacobSzwejbka

Differential Revision: D78500790

JacobSzwejbka commented:
@pytorchbot merge

JacobSzwejbka commented:
@pytorchmergebot merge -f "all workflows looks grean"
